google/cloud/bigquery/query.py

```diff
@@ -16,14 +16,21 @@
 
 from collections import OrderedDict
 import copy
-from typing import Union
+import datetime
+import decimal
+from typing import Optional, Union
 
 from google.cloud.bigquery.table import _parse_schema_resource
 from google.cloud.bigquery._helpers import _rows_from_json
 from google.cloud.bigquery._helpers import _QUERY_PARAMS_FROM_JSON
 from google.cloud.bigquery._helpers import _SCALAR_VALUE_TO_JSON_PARAM
 
 
+_SCALAR_VALUE_TYPES = Optional[
+    Union[str, int, float, decimal.Decimal, bool, datetime.datetime, datetime.date]
+]
```

Comment:
(nit) Shouldn't a singular form be used, since it's a definition of a single type? (that just happens to be an union of multiple types)